### PR TITLE
add algorithm to backtrack

### PR DESCRIPTION
**Pull request by vbyeh**
Added the solution to Word Search II on LeetCode under the backtrack directory

### CODE REVIEW

File: backtrack/word_search.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+         ['i','h','k','r'],
+         ['i','f','l','v']
+         ]
+
+words = ["oath","pea","eat","rain"]
+
+print findWords(board, words)
+'''
+
+def findWords(board, words):
```
Comment (keon) [inline backtrack/word_search.py]: let's use snake case

File: backtrack/word_search.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    if i < 0 or i >= len(board) or j < 0 or j >= len(board[0]):
+        return
+    
+    if not used[i][j] and board[i][j] in trie:
+        used[i][j]=True
+        find(board,i+1,j,trie[board[i][j]],pre+board[i][j], used, result)
+        find(board,i,j+1,trie[board[i][j]],pre+board[i][j], used, result)
+        find(board,i-1,j,trie[board[i][j]],pre+board[i][j], used, result)
+        find(board,i,j-1,trie[board[i][j]],pre+board[i][j], used, result)
+        used[i][j]=False
```
Comment (keon) [inline backtrack/word_search.py]: it'd be nice to have some test cases under if __name__ == "__main__".

File: backtrack/word_search.py [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+    result = set()
+    used = [[False]*len(board[0]) for _ in range(len(board))]
+    
+    for i in range(len(board)):
+        for j in range(len(board[0])):
+            find(board, i, j, trie, '', used, result)
+    return list(result)
+
+def find(board, i, j, trie, pre, used, result):
```
Comment (keon) [inline backtrack/word_search.py]: would 'backtract()' be a better name?

Comment (keon) [inline backtrack/word_search.py]: also could you explain briefly what trie, pre, used and result are? (with types)

**Comment by vbyeh** [comment]
thanks for the feedback! I changed a few things and added unittests

**Pull request merged**